src/mbgl/tile/geometry_tile.cpp

```diff
@@ -147,7 +147,7 @@ void GeometryTile::queryRenderedFeatures(
     const TransformState& transformState,
     const RenderedQueryOptions& options) {
 
-    if (!featureIndex || !data) return;
+    if (!featureIndex || !data || availableData != DataAvailability::All) return;
```

Comment:
Instead of returning an empty result set for queries made between layout and placement, I think we could effectively exclude symbol layers from the query by resetting `collisionTile` in `GeometryTile::onLayout`. Then checking the collision tile [will be skipped](https://github.com/mapbox/mapbox-gl-native/blob/bce352bd86f7faf98b3ba5bbb31d3a7860533199/src/mbgl/geometry/feature_index.cpp#L85-L88), but querying everything else is still possible.